tests/ops/op_math/test_evolution.py

```diff
@@ -83,6 +83,13 @@ def test_data(self):
         assert op.coeff == 1j * op.data[0]
         assert op.param == op.data[0]
 
+        new_param = np.array(2.345)
+        op.data = [new_param]
+
+        assert op.data == [new_param]
+        assert op.coeff == 1j * op.data[0]
+        assert op.data == op.data[0]
```

Comment:
Are there any tests for using the setter for invalid data? What would happen if I set `op.data = np.array([1.2, 2.3])`? Would the operator then be considered broadcasted or would it break something?

Reply:
I didn't add tests because we don't have a precedent for this. The rule is "don't do it".

Example:
```pycon
>>> op = qml.PauliX(0); op.data = 123; op.data
123
```

We can ensure the data is well-formed in all data setters if we feel it's important. Personally, I'd rather focus on making immutable operators.. but short-term this will be very helpful.